subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -1868,6 +2123,197 @@ int bt_mesh_cfg_mod_pub_set_vnd(uint16_t net_idx, uint16_t addr, uint16_t elem_a
+{
+	return mod_pub_va_set(net_idx, addr, elem_addr, mod_id, CID_NVAL,
+			      pub, status);
+}
+
+int bt_mesh_cfg_mod_pub_va_set_vnd(uint16_t net_idx, uint16_t addr,
+				   uint16_t elem_addr, uint16_t mod_id, uint16_t cid,
+				   struct bt_mesh_cfg_mod_pub_va *pub, uint8_t *status)
+{
+	if (cid == CID_NVAL) {
+		return -EINVAL;
+	}
+
+	return mod_pub_va_set(net_idx, addr, elem_addr, mod_id, cid, pub, status);
+}
+
+int bt_mesh_cfg_lpn_timeout_get(uint16_t net_idx, uint16_t addr,
+				uint16_t lpn_address, int32_t *timeout)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_LPN_TIMEOUT_STATUS, 2);
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = net_idx,
+		.app_idx = BT_MESH_KEY_DEV_REMOTE,
+		.addr = addr,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	struct lpn_timout_param param = {
+		.timeout = timeout,
+		.lpn_address = lpn_address,
+	};
+	int err;
+
+	err = cli_prepare(&param, OP_LPN_TIMEOUT_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_LPN_TIMEOUT_GET);
+	net_buf_simple_add_le16(&msg, lpn_address);
+
+	err = bt_mesh_model_send(cli->model, &ctx, &msg, NULL, NULL);
+	if (err) {
+		BT_ERR("model_send() failed (err %d)", err);
+		cli_reset();
+		return err;
+	}
+
+	return cli_wait();
+}
+
+int bt_mesh_cfg_krp_get(uint16_t net_idx, uint16_t addr,
+			uint16_t key_net_idx, uint8_t *status, uint8_t *phase)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_KRP_GET, 2);
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = net_idx,
+		.app_idx = BT_MESH_KEY_DEV_REMOTE,
+		.addr = addr,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	struct krp_param param = {
+		.status = status,
+		.phase = phase,
+		.net_idx = key_net_idx,
+	};
+	int err;
+
+	err = cli_prepare(&param, OP_KRP_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_KRP_GET);
+	net_buf_simple_add_le16(&msg, key_net_idx);
+
+	err = bt_mesh_model_send(cli->model, &ctx, &msg, NULL, NULL);
+	if (err) {
+		BT_ERR("model_send() failed (err %d)", err);
+		cli_reset();
+		return err;
+	}
+
+	if (!status) {
+		cli_reset();
+		return 0;
+	}
+
+	return cli_wait();
+}
+
+int bt_mesh_cfg_krp_set(uint16_t net_idx, uint16_t addr,
+			uint16_t key_net_idx, uint8_t val,
+			uint8_t *status, uint8_t *phase)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_KRP_SET, 3);
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = net_idx,
+		.app_idx = BT_MESH_KEY_DEV_REMOTE,
+		.addr = addr,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	struct krp_param param = {
+		.status = status,
+		.phase = phase,
+		.net_idx = key_net_idx,
+	};
+	int err;
+
+	err = cli_prepare(&param, OP_KRP_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_KRP_SET);
+	net_buf_simple_add_le16(&msg, key_net_idx);
+	net_buf_simple_add_u8(&msg, val);
+
+	err = bt_mesh_model_send(cli->model, &ctx, &msg, NULL, NULL);
+	if (err) {
+		BT_ERR("model_send() failed (err %d)", err);
+		cli_reset();
+		return err;
+	}
+
+	if (!status) {
+		cli_reset();
+		return 0;
+	}
+
+	return cli_wait();
+}
```

Comment:
Return error on invalid phases? (`phase <= 1 || phase > 3`)

Reply:
Can let the server model do this check?

Reply:
It will, but the API shouldn't allow illegal values to go on air